dispatchers.go

```diff
@@ -649,3 +651,16 @@ func errTooManyArguments(command string) error {
 func errNotJSON(command string) error {
 	return fmt.Errorf("%s requires the arguments to be in JSON form", command)
 }
+
+// makeUserArgs - Package the variables from the Dockerfile defined by
+// the ENV aand the ARG statements into one slice so the values
+// defined by both can later be evaluated when resolving variables
+// such as ${MY_USER}
+func makeUserArgs(b *Builder) (userArgs []string) {
+
+	userArgs = b.Env
+	for key, value := range b.Args {
+		userArgs = append(userArgs, key+"="+value)
```

Comment:
Don't know the answer to this, but how should we handle conflicts 
ARGS a=b
ENV a=c

Reply:
I contemplated that and didn't come up with a good answer.  My first thought was to check the existing values in the ENV list and not overwrite any of those.  But then I couldn't think of why ENV values would have precedence over ARG values in a Dockerfile.  So I took the chicken's way out and added them all leaving it up to the user to not step on themselves.

Happy to adjust if there's a better way.  I could potentially at least log a warning if I'm about to overwrite one.

Reply:
I'd be tempted to see how `docker build` interprets a variable when an `ARG` and an `ENV` set conflicting values for a given name, then add a conformance test that uses conflicting values.

Reply:
@TomSweeneyRedHat I agree with @nalind lets do what Docker does.

Reply:
Just ran with Docker, it ignores the ARG variable if there's a corresponding ENV variable.  I worked most of that code up over the weekend in my sandbox, I'll go see if I can wrap that up.